TombEngine/Game/effects/tomb4fx.cpp

```diff
@@ -371,31 +371,25 @@ void UpdateFireProgress()
 
 void AddFire(int x, int y, int z, short roomNum, float size, short fade)
 {
-	FIRE_LIST* fptr = &Fires[0];
-	int i = 0;
-	while (fptr->on)
-	{
-		fptr++;
-		if (++i >= MAX_FIRE_LIST)
-			return;
-	}	
+	FIRE_LIST newFire;
 	
 	if (fade)
-		fptr->on = fade;
+		newFire.on = fade;
 	else
-		fptr->on = 1;
+		newFire.on = 1;
+
+	newFire.x = x;
+	newFire.y = y;
+	newFire.z = z;
+	newFire.roomNumber = roomNum;
+	newFire.size = size;
 
-	fptr->x = x;
-	fptr->y = y;
-	fptr->z = z;
-	fptr->roomNumber = roomNum;
-	fptr->size = size;
+	Fires.push_back(newFire);
```

Comment:
The `Fires` vector shouldn't grow infinitely because effects need safeguards. Removing limits doesn't mean we literally remove them, rather we set *sensible* limits. :) If the legacy fire struct is still being used, you can write a special version of the `GetNewEffect()` function (found in `effects.h`) and call it here. When the vector has reached its maximum capacity and a new fire is being added, you could search for the smallest fire and overwrite it for example.